src/main/java/org/opentripplanner/graph_builder/linking/StreetSplitter.java

```diff
@@ -1,11 +1,834 @@
+        TraverseModeSet traverseModeSet,
+        RoutingRequest options,
+        final boolean destructiveSplitting,
+        boolean createSemiPermanentEdges
+    ) {
+        // find nearby street edges
+        // TODO: we used to use an expanding-envelope search, which is more efficient in
+        // dense areas. but first let's see how inefficient this is. I suspect it's not too
+        // bad and the gains in simplicity are considerable.
+        final double radiusDeg = SphericalDistanceLibrary.metersToDegrees(MAX_SEARCH_RADIUS_METERS);
+
+        Envelope env = new Envelope(vertex.getCoordinate());
+
+        // Perform a simple local equirectangular projection, so distances are expressed in degrees latitude.
+        final double xscale = Math.cos(vertex.getLat() * Math.PI / 180);
+
+        // Expand more in the longitude direction than the latitude direction to account for converging meridians.
+        env.expandBy(radiusDeg / xscale, radiusDeg);
+
+        final double DUPLICATE_WAY_EPSILON_DEGREES = SphericalDistanceLibrary.metersToDegrees(DUPLICATE_WAY_EPSILON_METERS);
+
+        // We sort the list of candidate edges by distance to the stop
+        // This should remove any issues with things coming out of the spatial index in different orders
+        // Then we link to everything that is within DUPLICATE_WAY_EPSILON_METERS of of the best distance
+        // so that we capture back edges and duplicate ways.
+        List<StreetEdge> candidateEdges = idx.query(env).stream()
+            .filter(
+                streetEdge -> streetEdge instanceof StreetEdge &&
+                    // do not find SemiPermanentPartialStreetEdges if creating new SemiPermanentPartialStreetEdges
+                    // use only permanent StreetEdges instead
+                    (createSemiPermanentEdges
+                        ? !(streetEdge instanceof SemiPermanentPartialStreetEdge)
+                        : true)
+            )
+            .map(edge -> (StreetEdge) edge)
+            // note: not filtering by radius here as distance calculation is expensive
+            // we do that below.
+            .filter(edge -> edge.canTraverse(traverseModeSet) &&
+                // only link to edges still in the graph.
+                edge.getToVertex().getIncoming().contains(edge))
+            .collect(Collectors.toList());
+
+        // Make a map of distances to all edges.
+        final TIntDoubleMap distances = new TIntDoubleHashMap();
+        for (StreetEdge e : candidateEdges) {
+            distances.put(e.getId(), distance(vertex, e, xscale));
+        }
+
+        // Sort the list.
+        Collections.sort(candidateEdges, (o1, o2) -> {
+            double diff = distances.get(o1.getId()) - distances.get(o2.getId());
+            // A Comparator must return an integer but our distances are doubles.
+            if (diff < 0)
+                return -1;
+            if (diff > 0)
+                return 1;
+            return 0;
```

Comment:
I'd like to see these return statements on the same line.

Reply:
Will compress

Reply:
I tried changing it to a ternary and it's much less readable. I'm just gonna keep it as is.